test/e2e/apimachinery/aggregator.go

```diff
@@ -550,6 +576,368 @@ func TestSampleAPIServer(f *framework.Framework, aggrclient *aggregatorclient.Cl
 	cleanTest(client, aggrclient, namespace)
 }
 
+// TestAPIService uses the sample-apiserver
+// (v1beta1.wardle.example.com) to test replace
+// status and deleteCollection APIService endpoints
+func TestAPIService(f *framework.Framework, aggrclient *aggregatorclient.Clientset, image string) {
+	ginkgo.By("Registering the sample API server.")
+	client := f.ClientSet
+	restClient := client.Discovery().RESTClient()
+
+	namespace := f.Namespace.Name
+	certCtx := setupServerCert(namespace, "beta-sample-api")
+
+	// kubectl create -f namespace.yaml
+	// NOTE: aggregated apis should generally be set up in their own namespace. As the test framework is setting up a new namespace, we are just using that.
+
+	// kubectl create -f secret.yaml
+	secretName := "beta-sample-apiserver-secret"
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+		},
+		Type: v1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"tls.crt": certCtx.cert,
+			"tls.key": certCtx.key,
+		},
+	}
+	_, err := client.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "creating secret %q in namespace %q", secretName, namespace)
+
+	// kubectl create -f clusterrole.yaml
+	_, err = client.RbacV1().ClusterRoles().Create(context.TODO(), &rbacv1.ClusterRole{
+
+		ObjectMeta: metav1.ObjectMeta{Name: "beta-sample-apiserver-reader"},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1helpers.NewRule("get", "list", "watch").Groups("").Resources("namespaces").RuleOrDie(),
+			rbacv1helpers.NewRule("get", "list", "watch").Groups("admissionregistration.k8s.io").Resources("*").RuleOrDie(),
+		},
+	}, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "creating cluster role %s", "sample-apiserver-reader")
```

Comment:
this means passing conformance requires enabling the RBAC API?

Reply:
This is already part of the current conformance test for [APIService endpoints](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/aggregator.go#L152-L155)

Reply:
cc @smarterclayton ... I thought RBAC was not supposed to be required for conformance tests to pass... if so, we shouldn't dig the hole deeper

Reply:
@liggitt @smarterclayton Checking the conformance meeting document I have found a bullet point "RBAC not for conformance" from 8 Oct 2019. I'm having issues tracking down the video for that meeting so I can't tell what was discussed or if that issue was ever followed up.

I'm wanting to better understand the historical context around RBAC so I can move this test forward with any insight that either you or @smarterclayton may have. TIA

p.s Nothing found on the sig-architecture mailing list around "conformance and RBAC"

Reply:
A side point about RBAC was made by @johnbelamaric at this [point in the meeting](https://www.youtube.com/watch?v=UhBwd_7vr_0&t=1226s) and was noted by @hh. The main discussion point the time was around some other changes that @smarterclayton had made.

Reply:
speculatively writing to the RBAC API to try to set up authorization rules for the test is ~ok, as long as RBAC API failures are tolerated

the fundamental thing this test should be checking is API calls to the APIServices API and the resulting behavior, which is independent of RBAC